### DISCUSSION TITLE
add_forceincludes doesn't generate ForcedIncludeFiles in VSXmake project files

### DISCUSSION THREAD

**Discussion by ifarbod** [xmake-io/xmake#5081] [Q&A] [answered]
I use `add_forceincludes` to add a common header between some projects in XMake. Building all of them works fine, but when I open the project inside Visual Studio 2022, IntelliSense doesn't recognize my types/macros in those force-included files, this is because XMake doesn't generate `<ForcedIncludeFiles>` inside the `vcxproj` for my projects.

```lua
add_forceincludes("common/macros.hpp")
```

```xml
  <Import Project="$(XmakeProgramDir)\scripts\vsxmake\vsproj\Xmake.props" />
  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='debug|x64'">
    <ClCompile>
      <ForcedIncludeFiles>common/macros.hpp;%(ForcedIncludeFiles)</ForcedIncludeFiles>
    </ClCompile>
```

**Comment by waruqi** [comment]
try this patch. https://github.com/xmake-io/xmake/pull/5084

```console
xmake update -s github:xmake-io/xmake#vsxmake
```

**Reply by ifarbod** [reply]
I have this in my config but IntelliSense doesn't recognize them, apparently, it doesn't like full paths (?)

![image](https://github.com/xmake-io/xmake/assets/4181752/60b8b606-c48d-435e-97ff-28a9f40f2255)

**Reply by waruqi** [reply, marked as answer]
try it again. `xmake update -s dev`

**Comment by ifarbod** [comment]
Thanks, that worked for me.

![image](https://github.com/xmake-io/xmake/assets/4181752/cb5803a6-1c35-47e8-86cc-b5812d55e623)
